extractor/filesystem/filesystem.go

```diff
@@ -165,7 +165,71 @@ func runOnScanRoot(ctx context.Context, config *Config, scanRoot *scalibrfs.Scan
 		return inventory.Inventory{}, nil, err
 	}
 
-	return RunFS(ctx, config, wc)
+	// Run extractors on the scan root
+	inv, status, err := RunFS(ctx, config, wc)
+	if err != nil {
+		return inv, status, err
+	}
+
+	// Process embedded filesystems
+	var additionalInv inventory.Inventory
+	for _, embeddedFS := range inv.EmbeddedFSs {
+		// Mount the embedded filesystem
+		mountedFS, err := embeddedFS.GetEmbeddedFS(ctx)
+		if err != nil {
+			status = append(status, &plugin.Status{
+				Name:    "EmbeddedFS",
+				Version: 1,
+				Status: &plugin.ScanStatus{
+					Status:        plugin.ScanStatusFailed,
+					FailureReason: fmt.Sprintf("failed to mount embedded filesystem %s: %v", embeddedFS.Path, err),
+				},
+			})
+			continue
+		}
+
+		// Create a new ScanRoot for the mounted filesystem
+		newScanRoot := &scalibrfs.ScanRoot{
+			FS:   mountedFS,
+			Path: "", // Virtual filesystem
+		}
+
+		// Run extractors on the mounted filesystem using Run
+		mountedInv, mountedStatus, err := Run(ctx, &Config{
```

Comment:
It might be better to reuse the existing config struct and just rewrite config.ScanRoots, config.PathsToExtract, etc.
We may add new fields to the struct over time and we may miss updating this code leading to hard-to-debug errors.

Reply:
Fair enough.